AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1724,6 +1720,40 @@ - (void)exitInterfaceState:(ASInterfaceState)interfaceState
   });
 }
 
+- (ASHierarchyState)hierarchyState
+{
+  ASDN::MutexLocker l(_propertyLock);
+  return _hierarchyState;
+}
+
+- (void)setHierarchyState:(ASHierarchyState)newState
+{
+  ASHierarchyState oldState;
+  {
+    ASDN::MutexLocker l(_propertyLock);
+    oldState = _hierarchyState;
+    _hierarchyState = newState;
+  }
+  
+  if (newState != oldState) {
+    LOG(@"setHierarchyState: oldState = %lu, newState = %lu", (unsigned long)oldState, (unsigned long)newState);
```

Comment:
Same as in `setInterfaceState`, although in this case there is not much difference because this block is short and clean.